src/components/swap/ConfirmSwapModal.tsx

```diff
@@ -261,6 +266,21 @@ export default function ConfirmSwapModal({
     startSwapFlow,
   ])
 
+  const l2Badge = useCallback(() => {
+    if (isL2ChainId(chainId) && confirmModalState !== ConfirmModalState.REVIEWING) {
+      const info = getChainInfo(chainId)
+      return (
+        <Badge>
+          <RowFixed data-testid="confirmation-modal-chain-icon" gap="sm">
+            <StyledLogo src={info.logoUrl} />
+            {info.label}
+          </RowFixed>
+        </Badge>
+      )
+    }
+    return undefined
```

Comment:
nit: this is not needed, `return undefined` is the default return value and I believe TypeScript will infer that automatically too

Reply:
this is needed to silence a linter complaint that "not all code paths return a value"

Reply:
`return` has type `void`, whereas `return undefined` has type `undefined`.